fmt: fix error priority, make goal-errors more helpful

BenWiederhake opened the pull request:
Fixes #6352.

In particular, this changes a deviation from GNU to a more helpful deviation from GNU:
```console
$ ../gnu/src/fmt -g apple
../gnu/src/fmt: invalid width: 'apple'
[$? = 1]
$ cargo run -q fmt -g apple # Before this PR
fmt: invalid value 'apple'
[$? = 1]
$ cargo run -q fmt -g apple # After this PR
fmt: invalid goal: 'apple'
[$? = 1]
```

cakebaker commented on tests/by-util/test_fmt.rs:
```diff
@@ -172,10 +187,42 @@ fn test_fmt_invalid_goal() {
+        .code_is(1)
+        .no_stdout()
+        .stderr_is("fmt: invalid width: 'banana'\n");
+    new_ucmd!()
+        .args(&["one-word-per-line.txt", "-w", "apple", "-g", "banana"])
+        .fails()
+        .code_is(1)
+        .no_stdout()
+        .stderr_is("fmt: invalid width: 'apple'\n");
```
This test is a bit confusing: while reading I wondered why you swapped the options (from `-g -w` to `-w -g`) but kept the values at the original positions. I think this distracts from what the test function is about: to show (and ensure) that an invalid width has a higher priority than an invalid goal.

My suggestion is to remove this test because the next test (`-w banana -g apple`) tests the same but in a more intuitive way.

BenWiederhake commented:
Changes since last push:
- Rebased
- Remove testcase of questionable use (thanks @cakebaker!)

In case anyone's curious, I was a bit paranoid about whether the value itself influences the priority in any way. This test proves that it does not; just as expected. That level of paranoia is not necessary with our own code.

cakebaker commented:
Thanks!
